fvtest/compilertest/compile/Method.cpp

```diff
@@ -67,8 +67,10 @@ ResolvedMethod::signature(TR_Memory * trMemory, TR_AllocationKind allocKind)
    {
    if( !_signature )
       {
-      char * s = (char *)trMemory->allocateMemory(strlen(_fileName) + 1 + strlen(_lineNumber) + 1 + strlen(_name) + 1, allocKind);
-      sprintf(s, "%s:%s:%s", _fileName, _lineNumber, _name);
+      char * s = (char *)trMemory->allocateMemory(_fileName.length() + 1 + strlen(_lineNumber) + 1 + _name.length() + 1, allocKind);
+
+      // TODO: Change this to use C++ string concatenation
+      sprintf(s, "%s:%s:%s", _fileName.c_str(), _lineNumber, _name.c_str());
```

Comment:
I am unsure if we can get rid of the explicit allocation, given that the lifetime of the (signature) string has to be longer than the lifetime of this function call (and therefore, the caller of this function is providing the alloc info). However, if it is known that the string will not be used / accessed beyond the lifetime of this `ResolvedMethod` instance, then we could just store it as a member variable, and return the corresponding c_str.

As far as the TODO is concerned, I can't think of a reason as to why we should use C++ string concat. Doing it using `sprintf` will likely be more efficient. I will remove it (the TODO).